packages/react/src/SegmentedControl/SegmentedControlIconButton.tsx

```diff
@@ -64,27 +68,56 @@ export const SegmentedControlIconButton: React.FC<React.PropsWithChildren<Segmen
         sxProp as SxProp,
       )
 
-  return (
-    <Box
-      as="li"
-      sx={mergedSx}
-      className={clsx(enabled && classes.Item, className)}
-      data-selected={selected || undefined}
-    >
-      {/* TODO: Once the tooltip remediations are resolved (especially https://github.com/github/primer/issues/1909) - bring it back */}
-      <SegmentedControlIconButtonStyled
-        aria-label={ariaLabel}
-        aria-current={selected}
-        sx={enabled ? undefined : getSegmentedControlButtonStyles({selected})}
-        className={clsx(enabled && classes.Button, enabled && classes.IconButton)}
-        {...rest}
+  if (unsafeDisableTooltip) {
+    return (
+      <Box
+        as="li"
+        sx={mergedSx}
+        className={clsx(enabled && classes.Item, className)}
+        data-selected={selected || undefined}
+      >
+        <SegmentedControlIconButtonStyled
+          aria-label={ariaLabel}
+          aria-current={selected}
+          sx={enabled ? undefined : getSegmentedControlButtonStyles({selected})}
+          className={clsx(enabled && classes.Button, enabled && classes.IconButton)}
+          {...rest}
+        >
+          <span className={clsx(enabled ? classes.Content : 'segmentedControl-content')}>
+            {isElement(Icon) ? Icon : <Icon />}
+          </span>
+        </SegmentedControlIconButtonStyled>
+      </Box>
+    )
+  } else {
+    return (
+      <Box
+        as="li"
+        sx={mergedSx}
+        className={clsx(enabled && classes.Item, className)}
+        data-selected={selected || undefined}
       >
-        <span className={clsx(enabled ? classes.Content : 'segmentedControl-content')}>
-          {isElement(Icon) ? Icon : <Icon />}
-        </span>
-      </SegmentedControlIconButtonStyled>
-    </Box>
-  )
+        <Tooltip
+          type={description ? undefined : 'label'}
+          text={description ? description : ariaLabel}
+          direction={tooltipDirection}
+        >
+          <SegmentedControlIconButtonStyled
+            aria-current={selected}
+            // If description is provided, we will use the tooltip to describe the button, so we need to keep the aria-label to label the button.
+            aria-label={description ? ariaLabel : undefined}
```

Comment:
Since several of the props are the same in line 79-89 should we make a `const`.

```tsx
const rest = {
  aria-current: selected,
  sx: enabled ? undefined : getSegmentedControlButtonStyles({selected}),
  className: clsx(enabled && classes.Button, enabled && classes.IconButton),
  ...rest
}
```

We could also add children ^.

Reply:
good thought! hmmmm this is a lot of duplication, but I'm thinking that it's maybe fine since we'll remove everything in the `else` in  1-2 weeks when we GA the code!